File: python/leetcode/150.py
class Solution:
    def evalRPN(self, tokens):
        if not tokens:
            return 0
        stack = []
        for item in tokens:
            if item in '+-/*':
                n1, n2 = stack.pop(), stack.pop()
                if item == '+':
                    stack.append(n1 + n2)
                elif item == '-':
                    stack.append(n2 - n1)
                elif item == "*":
                    stack.append(n1 * n2)
                else:
                    stack.append(n2 / n1)
            else:
                stack.append(int(item))
        return stack.pop()

File: cpp/leetcode/150.cpp
class Solution {
public:
    int evalRPN(vector<string> &tokens) {
        stack<int> res;
        for (string item : tokens) {
            if (item == "+" || item == "-" || item == "*" || item == "/") {
                int n1 = res.top();
                res.pop();
                int n2 = res.top();
                res.pop();
                if (item == "+") {
                    res.push(n2 + n1);
                } else if (item == "-") {
                    res.push(n2 - n1);
                } else if (item == "*") {
                    res.push(n2 * n1);
                } else if (item == "/") {
                    res.push(n2 / n1);
                }
            } else {
                res.push(stoi(item));
            }
        }
        return res.top();
    }
};